src/runtime_src/driver/xclng/tools/xbutil/xbutil.cpp

```diff
@@ -190,15 +191,16 @@ int main(int argc, char *argv[])
         return execv( std::string( path + "/xbflash" ).c_str(), argv );
     } /* end of call to xbflash */
 
-    if( std::string( argv[ 1 ] ).compare( "top" ) == 0 ) {
-        optind++;
-        return xcldev::xclTop(argc, argv);
-    }
     if( std::string( argv[ 1 ] ).compare( "validate" ) == 0 ) {
         optind++;
         return xcldev::xclValidate(argc, argv);
     }
 
+    if( std::string( argv[ 1 ] ).compare( "top" ) == 0 ) {
+        optind++;
+        return xcldev::xclTop(argc, argv);
+    }
+
```

Comment:
Please do not create temporary C++ strings just for doing strcmp. Please replace this code with something like the following:
if (std::strcmp(argv[1], "top") == 0) {

}

Reply:
@sonals I looked at the code and it seems like I did not change this code. I just cut it since I was working on something else and then paste it in a different line which is why it shows up as an addition, but I will go ahead and make the change for "top" as well as for "validate"